Testing pip install

slabasan opened the pull request.

tgamblin commented on hatchet/tests/dataframe_ops.py:
```diff
@@ -20,3 +20,6 @@ def test_filter(mock_graph_literal):
     assert len(filtered_gf.dataframe) == 9
+
+    filtered_gf = gf.filter(lambda x: x['name'].startswith('g'))
+    assert len(filtered_gf.dataframe) == 7
```
There should be more tests than this.  In particular -- this should test whether the names of rows in the filtered frame really do start with 'g'.  LIkewise the test above should assert that the times are actually what was filtered for

tgamblin commented:
@bhatele @slabasan: I don't think we should have to put inclusive time in the literals.  They should just take `time` and the method should have an option to calculate inclusive time.

ilumsden committed:
Adds a GitHub Action to test the PyPI releases on a regular schedule

ilumsden committed:
Tries push trigger instead of workflow_dispatch

ilumsden committed:
Tries to fix the temporary pull_request trigger

ilumsden committed:
Changes Hatchet versions to strings

ilumsden committed:
Adds some tweaks that seem to fix issues in local testing

ilumsden committed:
Changes the tables install to fetch version 3.5.2, since it was the last version to support Python 2.7

ilumsden committed:
Adds an install for numexpr when running on Python 2.7

ilumsden committed:
Splits up dependency install between Python 2.7 and Python 3

ilumsden committed:
Adds a test

ilumsden committed:
Tries moving to a different directory to run pip install hatchet

ilumsden committed:
Copies unit tests out of Hatchet repo to prevent pytest from running setup.py

ilumsden committed:
Sets fail-fast to false to prevent pip-install-tester from killing all jobs once a single job fails

slabasan committed:
update hatchet version, install from test pypi, update setuptools

- check setuptools version
- pip install with --extra-url
- upgrade cython
- bump version
